runtime/vm/jnimisc.cpp

```diff
@@ -168,6 +168,36 @@ getCurrentClassLoader(J9VMThread *currentThread)
 		} else {
 			/* use the sender (native method) classloader */
 			classLoader = J9_CLASS_FROM_METHOD(nativeMethod)->classLoader;
+#if JAVA_SPEC_VERSION >= 15
+			/* special case - if jdk/internal/loader/NativeLibraries.load(NativeLibraryImpl impl, String name, boolean isBuiltin, boolean isJNI)
+			 * is the current native method, use the class loader of "impl.fromClass".
+			 * This nativeMethod can't be cached cause HCR might make it invalid.
+			 */
+			if (J9VMJDKINTERNALLOADERNATIVELIBRARIES_LOAD_METHOD(currentThread->javaVM) == nativeMethod) {
+				J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+				vmFuncs->internalEnterVMFromJNI(currentThread);
+				/* The current native method has a NativeLibraryImpl instance as its first argument */
+				j9object_t nativeLibraryImplObject = (j9object_t)(currentThread->arg0EA[0]);
+				Trc_VM_findNativeLibrariesLoad_nativeMethod(currentThread, nativeMethod, nativeLibraryImplObject, classLoader);
+				/* Handle the object reference being redirected by the stack grower */
+				if (J9_ARE_ANY_BITS_SET((UDATA)nativeLibraryImplObject, 1)) {
+					nativeLibraryImplObject = *(j9object_t*)((UDATA)nativeLibraryImplObject - 1);
+				}
+				Assert_VM_notNull(nativeLibraryImplObject);
```

Comment:
This appears to be an argument, not a receiver, so the NULL check is probably more appropriate. If it could be removed entirely, that would be nice, but the check is better than the assert, since we will at least function predictably in that case.

Reply:
Sure, change it to `NULL` check instead though it is unlikely to be `NULL`, in that case, use native method `classloader` instead crash on a `NULL` object.